doc/Rust-to-LLGO-Migration-Guide.md

```diff
@@ -0,0 +1,332 @@
+```
+
+if everything is installed correctly, you will see the output like this (depending on your system):
+
+```SH
+-I/usr/local/include -L/usr/local/lib -l<lib_name>
+```
+
+### LLGO Package Mapping
+
+Map functions from the Rust library to an LLGO package, ensuring type consistency:
+
+- LLGoPackage
+
+Specify `LLGoPackage` and use `pkg-config` to find the location of the lib library.
+
+```go
+const (
+	LLGoPackage = "link: $(pkg-config --libs csv_wrapper); -lcsv_wrapper"
+)
+```
+
+- Type
+
+If you want to use variables inside the struct, you can add them accordingly.
+If it can't be represented directly or is not needed, it can be represented in the form `Unused []byte`, the length of the array is determined by its size, and if the struct is only used as a pointer, then the array length can be `0`.
+
+```go
+type Reader struct {
+	Unused [0]byte
+}
+
+// type Reader struct {
+//    Unused [8]byte
+// }
+```
+
+If we want to calculate the size of this structure, we can use the following C code:
+
+```C
+printf("%d\n", sizeof(csv_reader));
+```
+
+- Ordinary functions
+
+Ordinary functions can be mapped in the form of `//go:linkname`.
+
+```C
+csv_reader *csv_reader_new(const char *file_path);
+```
+
+After mapping:
+
+```go
+//go:linkname NewReader C.csv_reader_new
+func NewReader(file_path *c.Char) *Reader
+```
+
+- Method
+
+Methods need to be mapped in the form of `// llgo:link (*Receiver)`.
+
+```C
+void csv_reader_free(csv_reader *reader);
+
+const char *csv_reader_read_record(csv_reader *reader);
+```
+
+After mapping:
+
+We can extract the first parameter as Receiver:
+
+```go
+// llgo:link (*Reader).Free C.csv_reader_free
+func (reader *Reader) Free() {}
+
+// llgo:link (*Reader).ReadRecord C.csv_reader_read_record
+func (reader *Reader) ReadRecord() *c.Char { return nil }
+```
+
+- Function pointer
+
+If you use a function pointer, that is, declare the function as a type separately, you need to use `// llgo:type C`to declare it.
+
+```c
+typedef size_t (*hyper_io_read_callback)(void*, struct hyper_context*, uint8_t*, size_t);
+
+void hyper_io_set_read(struct hyper_io *io, hyper_io_read_callback func);
+```
+
+After mapping:
+
+```go
+// llgo:type C
+type IoReadCallback func(c.Pointer, *Context, *uint8, uintptr) uintptr
+
+// llgo:link (*Io).SetRead C.hyper_io_set_read
+func (io *Io) SetRead(callback IoReadCallback) {}
+```
+
+Or declare the function directly in the parameter.
+
+```go
+// llgo:link (*Io).SetRead C.hyper_io_set_read
+func (io *Io) SetRead(ioSetReadCb func(c.Pointer, *Context, *uint8, uintptr) uintptr) {}
+```
+
+### Writing Examples and README
+
+Finally, provide example code and a detailed README file to help users understand how to use the generated library.
+
+### Example Code
+
+You can find the migrated instance from [llgoexamples](https://github.com/goplus/llgoexamples), in the lib directory is the migrated Rust library, and in the rust directory, the migrated mapping file and go demo.
```

Comment:
```suggestion
You can find the migrated examples in the [llgoexamples](https://github.com/goplus/llgoexamples). The migrated Rust libraries are in the `lib` directory, and the migrated mapping files and Go demos are in the `rust` directory.
```

Reply:
English grammar mistakes are fine, but make sure to utilize inline code and proper capitalization. These are basic requirements.